chsdi/lib/validation/mapservice.py

```diff
@@ -235,3 +245,15 @@ def offset(self, value):
                 raise HTTPBadRequest('Please provide an integer as an offset parameter')
             else:
                 self._offset = int(value)
+
+    @limit.setter
+    def limit(self, value):
+        if value is not None:
+            if value.isdigit() and int(value) >= 0:
+                self._limit = int(value)
+            else:
+                raise HTTPBadRequest('Please provide a positive integer for the parameter limit')
+
+    @order.setter
+    def order(self, value):
+        self._order = value
```

Comment:
The only valid value for now is distance. So you could check that the same way. first `if value is not None` and then `if value == 'distance`. The rest is the bad params.